determined-ai/determined
# fix: Trials use their own trial state enum [DET-9639]

## Description

Changes the types used by APIs to return a Trial's state as a member of the Trial State enum

Previously this had been using Experiment's State enum, so `enrichTrialState` could return Queued/Starting/Pulling/Running states.
We can disentangle these with the new state options added to the Trial.State enum.

## Test Plan

- Tests pass
- On API responses to `/api/v1/experiments/120/trials` etc. the trials have a variety of states
- On API responses to `/api/v1/experiments-search`, the experiments have a `bestTrial` object with a state such as `STATE_COMPLETED`

If you want to see the "enriched" active states returned for a trial, you need to check the API quickly after launching a new experiment.
- Launch a test experiment (inside `examples/tutorials/core_api_pytorch_mnist`, run `det e create const.yaml ./`)
- Get the ID of the experiment, load `/api/v1/experiments/###/trials` and look at state value
- Refresh API request to see trial state changing from starting to running / completed / error

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
